### ISSUE TITLE
implies doesn't work if value is zero

### ISSUE THREAD

**Issue by lilleyse** [yargs/yargs#1203]
```js
return yargs
    .options({
        longitude: {
            type: 'number'
        },
        latitude: {
            type: 'number'
        }
    })
    .implies('longitude', 'latitude')
    .implies('latitude', 'longitude')
    .argv;
```

A command like `node command.js --longitude=0` proceeds when it should have produced an implication failure message.

**Comment by lilleyse** [comment]
Possibly related - a command like `node command.js --longitude=0 --latitude=10` fails:

```
Missing dependent arguments:
 latitude -> longitude
```

**Comment by bcoe** [comment]
thanks for the bug report. If you're so inclined, would happily take a PR for a failing test, or a fix 😄

**Comment by shadowspawn** [comment]
Thanks for the report. This should be fixed by #1985.

Feel free to open a new issue if it comes up again, with new information and renewed interest.
 
Thank you  for your contributions.
